fix: 백엔드-AI 서비스 간 데이터 구조 불일치 해결

choichanwoo001 opened the pull request:
<!-- 
아래 중 하나의 타입을 선택해 제목 앞에 붙여주세요.
feat: 새로운 기능 추가
fix: 버그 수정
refactor: 코드 리팩토링
docs: 문서 추가 및 수정
style: 포맷팅, 세미콜론 등 비즈니스 로직 영향 없는 변경
test: 테스트 코드 관련 작업
chore: 빌드/패키지 매니저 설정 등 기타 변경
-->

# 제목: 

---

## ✅ PR 종류
- [ ] feat (새로운 기능 추가)
- [x] fix (버그 수정)
- [ ] refactor (코드 리팩토링)
- [ ] docs (문서 추가 및 수정)
- [ ] style (비즈니스 로직 영향 없는 변경)
- [ ] test (테스트 코드 관련 작업)
- [ ] chore (빌드/설정 등 기타 변경)

---

## 📝 작업 내용
<!-- 이번 PR에서 작업한 내용을 간단히 정리 -->

---
- 템플릿 내용 필드명 통일: 'body' → 'templateContent'
- 템플릿 제목 필드명 통일: 'title' → 'templateTitle'
- 사용자 입력 필드명 통일: 'user_input' → 'userMessage'
- 변수 리스트 구조 통일: 배열 형태로 처리하되 내부에서 딕셔너리로 자동 변환
- 중복 데이터 전송 제거: template 내부와 별도 필드 중복 해결
- AI 검증 파이프라인에서 일관된 데이터 구조 사용 보장

수정된 파일:
- back/src/main/java/com/example/service/AIService.java
- ai/models/alimtalk_models.py
- ai/validators/constraint_validator.py

## 🔍 변경 이유
<!-- 왜 이 작업을 했는지, 해결하려는 문제는 무엇인지 설명 -->

---
ai, back 데이터 구조 불일치
## 📸 스크린샷 (선택)
<!-- UI 변경 시 스크린샷 첨부 -->
<img width="323" height="395" alt="image" src="https://github.com/user-attachments/assets/7c3d0162-5e92-48c5-ad64-725e6111e2c8" />
<img width="225" height="88" alt="image" src="https://github.com/user-attachments/assets/34db04ee-f294-4d88-a401-f95f833d9024" />

---

## ✅ 체크리스트
- [ ] 코드 동작 확인
- [ ] 관련 테스트 작성/수정
- [ ] 문서 업데이트
- [ ] 리뷰 반영

Commit:
# This is a combination of 3 commits.
# This is the 1st commit message:

"fix : 검증 로직 불필요한 코드 수정 및 json 수정."

# Conflicts:
#	ai/main.py
#	ai/models/alimtalk_models.py
#	ai/routers/alimtalk_routes.py
#	ai/services/alimtalk_service.py
#	ai/validators/constraint_validator.py
#	ai/validators/validator_pipeline.py

Commit:
"fix: 검증 샘플 데이터 삭제"

choichanwoo001 committed:
fix: 백엔드-AI 서비스 간 데이터 구조 불일치 해결

- 템플릿 내용 필드명 통일: 'body' → 'templateContent'
- 템플릿 제목 필드명 통일: 'title' → 'templateTitle'
- 사용자 입력 필드명 통일: 'user_input' → 'userMessage'
- 변수 리스트 구조 통일: 배열 형태로 처리하되 내부에서 딕셔너리로 자동 변환
- 중복 데이터 전송 제거: template 내부와 별도 필드 중복 해결
- AI 검증 파이프라인에서 일관된 데이터 구조 사용 보장

수정된 파일:
- back/src/main/java/com/example/service/AIService.java
- ai/models/alimtalk_models.py
- ai/validators/constraint_validator.py

choichanwoo001 committed:
fix: dict 구조 -> list로 전체 수정

ryumyang commented:
ai_routes.py 의 190,265 째줄 변수 추출 형태도  r'#\{([^}]+)\}'  이렇게 바꿔야 한다고 생각하긴 했는데 한번 봐주실 수 있나요?

choichanwoo001 committed:
fix: 변수명 #{} 로 통일

choichanwoo001 committed:
f-string 안에서만 #{{변수명}}으로 수정

choichanwoo001 committed:
fix: 템플릿 검증 API 데이터 구조 불일치 문제 해결

- 프론트엔드에서 variableList를 VariableDto 객체 배열로 전송하도록 수정
- 백엔드 AIService에서 AI 서버로 전송하는 요청 데이터 매핑 수정
- TemplateService에서 검증 요청 시 누락된 필드들 추가
- 403 Forbidden 오류 및 AxiosError 해결

변경사항:
* front/src/api/index.ts: validateTemplate 함수에서 변수 데이터 구조 수정
* back/.../AIService.java: validateTemplateWithFastAPI 요청 매핑 수정
* back/.../TemplateService.java: AI 서버 요청 시 필수 필드 추가

choichanwoo001 committed:
fix: 템플릿 검증 API 데이터 구조 및 타임아웃 문제 해결

- 프론트엔드 API 타임아웃을 30초로 증가 (AI 검증 시간 고려)
- 백엔드에서 AI 서버로 전송하는 요청 데이터 매핑 수정
- TemplateService에서 검증 요청 시 누락된 필드들 추가
- validateTemplate 함수에서 VariableDto 객체 배열로 데이터 구조 통일

choichanwoo001 committed:
feat: AI 서버 한국어 카테고리 지원 및 비동기 처리 개선

- AlimtalkTemplate에서 한국어 카테고리 enum으로 변경
- 예약취소, 주문, 결제, 배송 등 다양한 한국어 카테고리 지원
- OpenAI 서비스를 AsyncOpenAI로 변경하여 진짜 병렬 처리 구현
- 검증 속도 개선: 17초 → 4-6초 (병렬 실행)

choichanwoo001 committed:
feat: AI 서버 검증 실패 시 상세 오류 정보 제공 개선

- 검증 실패 응답을 백엔드 호환 형식으로 변환
- 각 오류별 수정 제안 자동 생성 기능 추가
- 검증 단계 한국어 변환 (constraint → 1차 검증)
- validation_results 필드 추가로 백엔드 호환성 확보

choichanwoo001 committed:
feat: 템플릿 검증 실패 시 사용자 경험 개선

- 검증 실패 알림 메시지를 사용자 친화적으로 변경
- '검증 실패' → '템플릿 수정이 필요합니다' 안내
- 변수가 없는 경우에도 사이드바 표시하도록 개선
- 템플릿 전체 오류 처리 로직 추가

choichanwoo001 committed:
feat: 반려 사유 사이드바 UX 완전 개편

- 뒤로가기 버튼 추가로 직관적인 네비게이션 구현
- 반려된 변수 목록 제거 (의미 없는 정보)
- 오류 클릭 시 맞춤형 대안 3개 표시 기능
- 각 오류 유형별 구체적인 해결 방안 제시
- TypeScript 타입 안전성 강화

choichanwoo001 committed:
feat: 알림톡 미리보기에 문제 영역 시각적 하이라이트 추가

- 검증 오류 발생 시 템플릿 전체를 빨간색 테두리로 하이라이트
- 펄스 애니메이션으로 문제 영역 강조 효과
- 템플릿 전체 오류와 변수별 오류 구분 표시
- validationErrors prop 추가로 오류 정보 연동

choichanwoo001 committed:
fix: 변수명 형식때메 프롬프트 내용인데도 변수로 인식되는 오류

choichanwoo001 committed:
Merge branch 'develop' into fix/validate-date-transfer

Commit:
"fix: 검증 파이프라인 vue,back,ai 데이터구조 맞추는 중"

Commit:
"fix: 생성 결과에 맞게 검증 데이터 구조 변경"

choichanwoo001 committed:
refactor: ConstraintValidator에서 변수 추출 함수 제거 및 외부 변수 리스트 사용

- _extract_variables_from_template() 메서드 삭제
- _check_variable_usage_rules_async()에서 template_data.get('detected_variables') 사용
- validate() 메서드 docstring에 detected_variables 파라미터 설명 추가
- 변수 추출 로직 중복 제거로 효율성 향상

choichanwoo001 committed:
fix: 충돌 해결

choichanwoo001 committed:
chore: 삭제한 메서드 참조 삭제

choichanwoo001 committed:
docs: validators 폴더 내 파라미터명 일관성 및 문서화 개선

- validator_pipeline.py docstring에 detected_variables 파라미터 설명 추가
- prompts/variable_usage_prompt.py에서 variable_list → variable_names로 파라미터명 통일
- 프롬프트 템플릿 내 변수명도 일치하도록 수정
- validators 폴더 전체 파라미터명 일관성 확보

choichanwoo001 committed:
fix: CI 변수명 잘못 인식되는거 수정

ryumyang reviewed (commented):
고생하셨습니다. 검증 고도화 화이팅
